scheduler/core/programprovider/ocs/__init__.py

```diff
@@ -42,8 +42,13 @@ def ocs_program_data(program_list: Optional[bytes] = None) -> Iterable[dict]:
             list_file = program_list
         else:
             list_file = DEFAULT_PROGRAM_ID_PATH
-        with list_file.open('r') as file:
-            id_frozenset = frozenset(line.strip() for line in file if line.strip())
+
+        if isinstance(program_list, bytes):
+            file = program_list.decode('utf-8')
+            id_frozenset = frozenset(f.strip() for f in file.split('\n') if f.strip())
```

Comment:
This could be improved to reduce the calls to `f.strip()` by doing:
```python
id_frozenset = frozenset(fs for f in file.split('\n') if (fs := f.strip()))
```
Same analogously with the below `id_frozenset`.